jina/drivers/dump.py

```diff
@@ -0,0 +1,158 @@
+    :param args: passed to super().__init__
+    :param kwargs: passed to super().__init__
+    """
+
+    def __init__(
+        self,
+        executor: Optional[str] = None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(executor, 'dump', *args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+        """Call the Dump method of the Indexer to which the Driver is attached
+
+        :param args: passed to the exec_fn
+        :param kwargs: passed to the exec_fn
+        """
+        self.exec_fn(self.req.path, self.req.shards, self.req.formats, *args, **kwargs)
+
+
+class DumpPersistor:
+    """
+    Class for creating and importing from dumps
+
+    Do NOT instantiate. Only provides static methods
+    """
+
+    @staticmethod
+    def export_dump_streaming(
+        path: str,
+        shards: int,
+        size: int,
+        data: Iterable[Tuple[str, np.array, bytes]],
+    ):
+        """Export the data to a path, based on sharding,
+
+        :param path: path to dump
+        :param shards: the nr of shards this pea is part of
+        :param size: total amount of entries
+        :param data: the generator of the data (ids, vectors, metadata)
+        """
+        if os.path.exists(path):
+            raise Exception(f'path for dump {path} already exists. Not dumping...')
+        size_per_shard = size // shards
+        extra = size % shards
+        shard_range = list(range(shards))
+        for shard_id in shard_range:
+            if shard_id == shard_range[-1]:
+                size_this_shard = size_per_shard + extra
+            else:
+                size_this_shard = size_per_shard
+            shard_path = os.path.join(path, str(shard_id))
+            shard_written = 0
+            os.makedirs(shard_path)
+            vectors_fh, metas_fh, ids_fh = DumpPersistor._get_file_handlers(shard_path)
+            while shard_written < size_this_shard:
+                id_, vec, meta = next(data)
+                vec_bytes = vec.tobytes()
+                vectors_fh.write(
+                    len(vec_bytes).to_bytes(BYTE_PADDING, sys.byteorder) + vec_bytes
+                )
+                metas_fh.write(len(meta).to_bytes(BYTE_PADDING, sys.byteorder) + meta)
+                ids_fh.write(id_ + '\n')
+                shard_written += 1
+            vectors_fh.close()
+            metas_fh.close()
+            ids_fh.close()
+
+    @staticmethod
+    def import_vectors(path: str, pea_id: str):
+        """Import id and vectors
+
+        :param path: the path to the dump
+        :param pea_id: the id of the pea (as part of the shards)
+        :return: the generators for the ids and for the vectors
+        """
+        path = os.path.join(path, pea_id)
+        ids_gen = DumpPersistor._ids_gen(path)
+        vecs_gen = DumpPersistor._vecs_gen(path)
+        return ids_gen, vecs_gen
+
+    @staticmethod
+    def import_metas(path: str, pea_id: str):
```

Comment:
Are all these methods so generic for every potential DBMBSIndexer?

Reply:
We'll adapt as we add new ones, if required. But I think so, for now.